pkg/ottl/ottlfuncs/func_is_map_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func Test_IsMap(t *testing.T) {
+	tests := []struct {
+		name     string
+		value    interface{}
+		expected bool
+	}{
+		{
+			name:     "map",
+			value:    make(map[string]any, 0),
+			expected: true,
+		},
+		{
+			name:     "ValueTypeMap",
+			value:    pcommon.NewValueMap(),
+			expected: true,
+		},
+		{
+			name:     "not map",
+			value:    "not a map",
+			expected: false,
+		},
+		{
+			name:     "ValueTypeSlice",
+			value:    pcommon.NewValueSlice(),
+			expected: false,
+		},
+		{
+			name:     "nil",
+			value:    nil,
+			expected: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			exprFunc := isMap[any](&ottl.StandardPMapGetter[any]{
+				Getter: func(context.Context, interface{}) (interface{}, error) {
+					return tt.value, nil
+				},
+			})
+			result, err := exprFunc(context.Background(), nil)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func Test_IsMap_Error(t *testing.T) {
```

Comment:
We need one more case for a non-`TypeError` error so we cover all switch cases.

Reply:
Wait, I'm just seeing these will return a wrapped error now, good thinking with having these return a `TypeError`. If we have the nested getter return an invalid type that should cover all the cases.

Reply:
This is technically the non-TypeError case.  When this Getter returns an error the the StandardPMapGetter `Get` function sees it and returns it as a WrappedError.  When isMap gets that error it returns false and the error.

The case of the Getter returning a TypeError and then `false, nil` being returned is handled by the `not map` and `nil` cases above.

Reply:
Yeah, I got mixed up for a moment. The tests look good to me, thanks.